ebpf: Support configuring length of big fields for gadgets (#4791)
inspektor-gadget/inspektor-gadget: docs/gadget-devel/gadget-ebpf-api.md

- `ebpf.formatter.enum`: Name of the new field. If the annotation is not set and the source field name has a `_raw` suffix, the target name will be set to the source name without that suffix.
- `ebpf.formatter.bitfield.separator`: Separator used. Defaults to `|`.

## Flexible Strings

> Some gadgets may need to use strings whose maximum size can be configured at
> runtime to save memory. For that, gadgets can use the `gadget_flex_string`

Reviewer: ```suggestion
For some gadgets, you may want to use strings whose maximum size is configurable at
runtime to save memory. For that, gadgets can use the `gadget_flex_string`
```

type:

```c
#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
#pragma clang attribute push(__attribute__((preserve_access_index)), \
			     apply_to = record)

> #endif

Reviewer: Please add a comment that can help understanding what these lines do. Does it affect other fields that are not `gadget_flex_string`? Can we help with this using a macro (e.g. `gadget_flex_string(mystring, MAX_SIZE)`)?

struct event {
	gadget_flex_string mystring[MAX_SIZE];
};

#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
#pragma clang attribute pop
#endif
```

:::warning

The `gadget_flex_string` fields must be on the root of the event structure.

:::

The maximum size of the string can be set by the user at runtime using the
`<fieldName>-size` parameter. For example, to set the maximum size of `mystring`

> to 64 bytes, the user can run:

Reviewer: Sounds like this could become ambiguous when multiple structs use the same field name for gadget_flex_string. Now that we have the custom params: what about using an annotation instead (probably leads to a hen/egg problem, though)?

Reviewer: Yes - something like `--long-names` or `--short` as a parameter that would set the annotation to on that field. We keep the flexibility of resizing freely (using the annotations, for advanced users) while still getting a good UX for the normal user.
